emission/tests/storageTests/TestTimeSeries.py

```diff
@@ -81,6 +81,22 @@ def testExtraQueries(self):
         with self.assertRaises(AttributeError):
             list(ts.find_entries(time_query=tq, extra_query_list=[ignored_phones]))
 
+    def testCountData(self):
+        '''
+        Test 1 : Specific key with empty extra_queries
+        key = 'background/location', extra_query_list = []
+        Results in empty query = {}, which matches all documents for a user for that key.
+        Hence should return total count of all documents matching that key.
+        Testing this with sample dataset: "shankari_2015-aug-27"
+        '''
+        ts = esta.TimeSeries.get_time_series(self.testUUID)
+        total_count = ts.count_data("background/location",[])
+        print(total_count)
+        self.assertEqual(total_count, 555)
```

Comment:
How do you know that 555 is correct?

Reply:
With reference to my [notes on this issue](https://github.com/e-mission/e-mission-docs/issues/933#issuecomment-1698041613), I realized that the sample dataset `shankari_2015-aug-27` is the dataset on which the count functionality is tested in the test scripts.

So, I did a manual count of the number of occurrences of the key values: `"background/location"` and `"background/filtered_location"` using the grep commands:
- `grep -c "background/location" shankari_2015-aug-27.json`
- `grep -c "background/filtered_location" shankari_2015-aug-27.json`
- This returned the counts as 555 and 327, respectively.

I did assume the count is required for input data which is the one that gets stored in timeseries_db() or analysis_timeseries_db().

**If this approach is indeed correct, I can add the test input for "background/filtered_location" as well.**

Attached is the screenshot for the grep results.

<img width="1792" alt="Screen Shot 2023-08-31 at 11 19 52 AM" src="https://github.com/e-mission/e-mission-server/assets/37911144/0915a28a-cbd2-435b-ba07-a48e2c61cc07">

Reply:
This is a fine approach, but:
1. screenshots for code are bad because they are not searchable
2. put the justification *in the code*. How is somebody looking at this code 5 years later going to know how you validated the answer?

Reply:
Also, 

> I realized that the sample dataset shankari_2015-aug-27 is the dataset on which the count functionality is tested in the test scripts.

is incorrect. That is a suggestion on loading data for **manual** end-to-end testing. You cannot use manual instructions for automated testing - the test has to be self-contained and load all the data that it needs. This set of tests does load its own data already but not from that file.